sw-emulator/lib/periph/src/ml_dsa87.rs

```diff
@@ -0,0 +1,1254 @@
+}
+
+impl MlDsa87 {
+    /// NAME0 Register Value TODO update when known
+    const NAME0_VAL: RvData = 0x73656370; //0x63737065; // secp
+
+    /// NAME1 Register Value TODO update when known
+    const NAME1_VAL: RvData = 0x2D333834; // -384
+
+    /// VERSION0 Register Value TODO update when known
+    const VERSION0_VAL: RvData = 0x30302E31; // 1.0
+
+    /// VERSION1 Register Value TODO update when known
+    const VERSION1_VAL: RvData = 0x00000000;
+
+    pub fn new(clock: &Clock) -> Self {
+        Self {
+            name0: ReadOnlyRegister::new(Self::NAME0_VAL),
+            name1: ReadOnlyRegister::new(Self::NAME1_VAL),
+            version0: ReadOnlyRegister::new(Self::VERSION0_VAL),
+            version1: ReadOnlyRegister::new(Self::VERSION1_VAL),
+            control: ReadWriteRegister::new(0),
+            status: ReadOnlyRegister::new(Status::READY::SET.value),
+            iv: Default::default(),
+            seed: Default::default(),
+            sign_rnd: Default::default(),
+            message: Default::default(),
+            verification_result: Default::default(),
+            sk_out: [0; 1224],
+            sk_in: [0; 1224],
+            pk: [0; 648],
+            signature: [0; 1157],
+            timer: Timer::new(clock),
+            op_complete_action: None,
+        }
+    }
+
+    fn read_access_fault(&self, _size: RvSize, _index: usize) -> Result<RvData, BusError> {
+        Err(BusError::LoadAccessFault)
+    }
+
+    fn write_access_fault(
+        &self,
+        _size: RvSize,
+        _index: usize,
+        _val: RvData,
+    ) -> Result<(), BusError> {
+        Err(BusError::StoreAccessFault)
+    }
+
+    // Clear registers
+    fn zeroize(&mut self) {}
+
+    /// On Write callback for `control` register
+    ///
+    /// # Arguments
+    ///
+    /// * `size` - Size of the write
+    /// * `val` - Data to write
+    ///
+    /// # Error
+    ///
+    /// * `BusError` - Exception with cause `BusError::StoreAccessFault` or `BusError::StoreAddrMisaligned`
+    pub fn on_write_control(&mut self, size: RvSize, val: RvData) -> Result<(), BusError> {
+        // Writes have to be Word aligned
+        if size != RvSize::Word {
+            Err(BusError::StoreAccessFault)?
+        }
+
+        // Set the control register
+        self.control.reg.set(val);
+
+        match self.control.reg.read_as_enum(Control::CTRL) {
+            Some(Control::CTRL::Value::GEN_KEY)
+            | Some(Control::CTRL::Value::SIGN)
+            | Some(Control::CTRL::Value::VERIFY) => {
+                // Reset the Ready and Valid status bits
+                self.status
+                    .reg
+                    .modify(Status::READY::CLEAR + Status::VALID::CLEAR);
+
+                self.op_complete_action = Some(self.timer.schedule_poll_in(ML_DSA87_OP_TICKS));
+            }
+            _ => {}
+        }
+
+        if self.control.reg.is_set(Control::ZEROIZE) {
+            self.zeroize();
+        }
```

Comment:
Should this be part of the `match`?

Reply:
Not possible as the match is on a only a part of the